fms/models/llava_next.py

```diff
@@ -0,0 +1,603 @@
+import logging
+import math
+import re
+from dataclasses import dataclass, field
+from typing import Any, Mapping, Optional, Unpack, Tuple
+
+import torch
+import torch.nn as nn
+
+from fms import models
+from fms.models.siglip_vision import SiglipVision, SiglipVisionConfig
+from fms.models.granite import Granite, GraniteConfig
+from fms.modules.linear import get_linear_type
+from fms.modules.attention import AttentionKwargs
+from fms.distributed.strategy import DistributedStrategy, NoOpStrategy
+from fms.utils import serialization
+from fms.utils.activation import str_to_activation
+from fms.utils.config import ModelConfig
+
+# TODO: percolate ditributed_strategy, taking care of _no_split_modules from original transformers code
+
+logger = logging.getLogger(__name__)
+
+_granite_3_2_2b_text_config = GraniteConfig(
```

Comment:
Can you move this and the vision config to the end of the file with the example config for the whole model?